helm/resource_release.go

```diff
@@ -856,14 +856,14 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
 	}
 
 	if !useChartVersion(d.Get("chart").(string), d.Get("repository").(string)) {
```

Comment:
```suggestion
	} else if !useChartVersion(d.Get("chart").(string), d.Get("repository").(string)) {
```

We can skip this block if we already called `d.SetNewComputed("metadata")`.

Reply:
This makes sense to include.